onnxruntime/core/framework/bfc_arena.cc

```diff
@@ -42,22 +42,8 @@ BFCArena::BFCArena(std::unique_ptr<IAllocator> resource_allocator,
   stats_.bytes_limit = static_cast<int64_t>(total_memory);
 
   arena_extend_strategy_ = arena_extend_strategy;
+  UpdateFirstAllocationShrinkageLogic();
 
```

Comment:
Why not call and remove these lines?
SetArenaExtendStrategy(arena_extend_strategy)